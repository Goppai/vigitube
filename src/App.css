body {
  margin: 0;
  padding: 10px;
  font-family: sans-serif;
}

.width-100 {
  width: 100%;
}

.center-text {
  text-align: center;
}

/* SearchBar */

.search-bar {
  margin-bottom: 10px;
  display: flex;
}

.vigitube {
  margin: 0 10px;
  font-size: 28px;
  font-family: monospace;
}

.github-repo-link {
  text-align: right;
  margin: 0 30px 0 20px;
  font-size: 28px;
}

@media (max-width: 425px) {
  .search-bar {
    flex-wrap: wrap;
  }

  .vigitube {
    margin: 0 0 10px 0;
    padding-left: 4px;
    width: 50%;
  }

  .github-repo-link {
    width: 50%;
    margin: 0;
  }
}

.vigitube-part-one {
  transition: color .3s ease;
}

.vigitube-part-two {
  color: deepskyblue;
  transition: color .3s ease;
}

.vigitube:hover .vigitube-part-one {
  color: deepskyblue;
}

.vigitube:hover .vigitube-part-two {
  color: indianred;
}

/* VideoView */

.video-view {
  padding: 0 0 10px 0;
}

@media (min-width: 992px) {
  .video-view {
    padding: 0 10px 10px 0;
  }
}

/* VideoList */

.video-list {
  margin: 0 0 10px 0;
  border: 1px solid #e8e8e8;
  border-radius: 4px;
}

/* VideoListItem */

.video-list-item {
  cursor: pointer;
  padding: 10px;
  border: 0;
  transition: background-color .3s ease;
}

.video-list-item:hover {
  background-color: #eee;
}

/* Footer */

.footer {
  font-family: monospace;
}

.footer span {
  font-size: 18px;
}

/* overrides */
.ant-card-bordered {
  border-radius: 4px !important;
}

.ant-layout-footer {
  background: #fff !important;
}
