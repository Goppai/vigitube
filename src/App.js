import React, { Component } from 'react'
import { Row, Icon } from 'antd'
import { SearchBar, VideoView, VideoList, Footer } from './components'
import { search } from './utilities/functions'
import './App.css'

class App extends Component {
  constructor() {
    super()
    this.state = {
      videos: [],
      selectedVideo: null
    }
  }

  componentDidMount() {
    search().then(videos => {
      this.setState({
        videos,
        selectedVideo: videos[0]
      })
    })
  }

  render() {
    const { videos, selectedVideo } = this.state
    return (
      <Row className='width-100'>
        <SearchBar
          search={query => search(query)
            .then(videos => this.setState({
              videos,
              selectedVideo: videos[0]
            }))
          }
        />
        {selectedVideo && <VideoView selectedVideo={selectedVideo}/>}
        {videos && <VideoList selectVideo={selectedVideo => this.setState({ selectedVideo })} videos={videos}/>}
        <Footer>
          <span>Powered by <a href='https://www.youtube.com/' target='_blank' rel='noopener noreferrer'><Icon type='youtube'/></a></span>
        </Footer>
      </Row>
    )
  }
}

export default App
