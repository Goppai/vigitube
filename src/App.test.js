import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import App from './App'

describe('<App/>', () => {
  const wrapper = shallow(<App/>)

  it('renders', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('has intial state set correctly', () => {
    expect(wrapper.state().videos.length).toBe(0)
    expect(wrapper.state().selectedVideo).toBeNull()
  })
})
