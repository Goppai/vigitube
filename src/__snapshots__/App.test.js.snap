// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders 1`] = `
<Row
  className="width-100"
  gutter={0}
>
  <Component
    search={[Function]}
  />
  <Component
    selectVideo={[Function]}
    videos={Array []}
  />
  <Component>
    <span>
      Powered by 
      <a
        href="https://www.youtube.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <Icon
          type="youtube"
        />
      </a>
    </span>
  </Component>
</Row>
`;
